onadata/apps/logger/models/xform.py

```diff
@@ -822,6 +825,17 @@ def _set_encrypted_field(self):
             else:
                 self.encrypted = False
 
+    def _set_public_key_field(self):
+        if self.json and self.json != '':
+            if self.submission_count() == 0:
+                json_dict = json.loads(self.json)
+                json_dict['public_key'] = self.public_key
+                json_dict['submission_url'] = ODK_SUBMISSION_URL
```

Comment:
Submission URL is no longer a requirement for encrypted forms, I don't think we should enforce this here. We should remove it.

Reply:
Removed the `ODK_SUBMISSION_URL` in latest commits